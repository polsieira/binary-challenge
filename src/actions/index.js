export const checkIsLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
})

export const addExoplanets = (exoplanets) => ({
  type: 'ADD_EXOPLANETS',
  exoplanets
})

export const loginUser = (userInfo) => ({
  type: 'LOGIN_USER',
  userInfo
})

export const addFilters = (filters) => ({
  type: 'ADD_FILTERS',
  filters
})