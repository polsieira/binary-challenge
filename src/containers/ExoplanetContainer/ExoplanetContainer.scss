.ExoplanetContainer {
  background: white;
  border-radius: 5px;
  box-shadow: 2px 2px 2px rgb(211, 211, 211);
  display: inline-block;
  height: 50vh;
  overflow: scroll;
  width: 97%;

  .loading-message {
    font-size: 100px;
  }

  ul {
    align-items: center;
    display: flex;
    height: 50px;
    justify-content: space-around;
    margin: 0px;
    padding: 0px;

    li:first-child {
      border-left: none;
    }

    li {
      align-items: center;
      border-top: rgb(211, 211, 211) solid 2px;
      border-bottom: rgb(211, 211, 211) solid 2px;
      border-left: rgb(211, 211, 211) solid 2px;
      display: flex;
      height: 100%;
      justify-content: center;
      width: 20%;
    }
  }
}