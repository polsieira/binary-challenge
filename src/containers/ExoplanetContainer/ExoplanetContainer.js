import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import './ExoplanetContainer.scss';
import { checkIsLoading } from '../../actions';
import ExoplanetCard from '../ExoplanetCard/ExoplanetCard';
import { filterExoplanets } from '../../filterExoplanets';

export const ExoplanetContainer = ({ exoplanets, checkIsLoading, isLoading, filters }) => {
  checkIsLoading(true);
  const { distance, gravity, year } = filters;
  const filteredExoplanets = filterExoplanets(exoplanets, distance, gravity, year)
  const exoplanetCards = filteredExoplanets.map(({ pl_name, pl_hostname, pl_orbper, gaia_dist, pl_disc }) => {
    return (
      <ExoplanetCard
        key={pl_name}
        planetName={pl_name}
        hostStarName={pl_hostname}
        orbitalPeriod={pl_orbper}
        distance={gaia_dist}
        discoveryYear={pl_disc}
      />
    )
  })
  checkIsLoading(false);

  return (
    <div className="ExoplanetContainer">
      {isLoading && <p className="loading-message">Loading exoplanets</p>}
      <ul>
        <li>Planet Name</li>
        <li>Local Star</li>
        <li>Length of Year</li>
        <li>Distance</li>
        <li>Discovery Year</li>
      </ul>
      {exoplanetCards}
    </div>
  )

}

export const mapStateToProps = ({ exoplanets, isLoading, filters }) => ({
  exoplanets,
  isLoading,
  filters
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    checkIsLoading
  }, dispatch)
)


export default connect(mapStateToProps, mapDispatchToProps)(ExoplanetContainer);
