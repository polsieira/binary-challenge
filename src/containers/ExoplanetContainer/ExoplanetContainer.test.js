import React from 'react';
import { shallow } from 'enzyme';
import { ExoplanetContainer, mapStateToProps, mapDispatchToProps } from './ExoplanetContainer';
import { checkIsLoading } from '../../actions';

describe('ExoplanetContainer', () => {
  describe('ExoplanetContainer component', () => {
    let wrapper;
    const mockCheckIsLoading = jest.fn();
    const mockIsLoading = false;
    const mockExoplanets = [{
      pl_name: 'exoplanet',
      pl_hostname: 'star',
      pl_orbper: 2000,
      gaia_dist: 100,
      pl_disc: 2005
    }]

    beforeEach(() => {
      wrapper = shallow(<ExoplanetContainer exoplanets={mockExoplanets} checkIsLoading={mockCheckIsLoading} isLoading={mockIsLoading} />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when isLoading', () => {
      const mockIsLoading = true;

      wrapper = shallow(<ExoplanetContainer exoplanets={mockExoplanets} checkIsLoading={mockCheckIsLoading} isLoading={mockIsLoading} />);

      expect(wrapper).toMatchSnapshot();
    });

  });

  describe('mapStateToProps', () => {
    it('should return an object with exoplanets', () => {
      const mockState = {
        exoplanets: [{
          pl_name: 'exoplanet',
          pl_hostname: 'star',
          pl_orbper: 2000,
          gaia_dist: 100,
          pl_disc: 2005
        }],
        error: '',
        isLoading: false
      };

      const expected = {
        exoplanets: [{
          pl_name: 'exoplanet',
          pl_hostname: 'star',
          pl_orbper: 2000,
          gaia_dist: 100,
          pl_disc: 2005
        }],
        isLoading: false
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {

    const mockDispatch = jest.fn();

    it('should call dispatch with a checkIsLoading action when rendered', () => {
      const actionToDispatch = checkIsLoading(true);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.checkIsLoading(true);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });


  });

});