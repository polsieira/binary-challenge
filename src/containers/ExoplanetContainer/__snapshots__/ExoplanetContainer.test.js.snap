// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExoplanetContainer ExoplanetContainer component should match the snapshot 1`] = `undefined`;

exports[`ExoplanetContainer ExoplanetContainer component should match the snapshot when isLoading 1`] = `
<div
  className="ExoplanetContainer"
>
  <p
    className="loading-message"
  >
    Loading exoplanets
  </p>
  <ul>
    <li>
      Planet Name
    </li>
    <li>
      Local Star
    </li>
    <li>
      Length of Year
    </li>
    <li>
      Distance
    </li>
    <li>
      Discovery Year
    </li>
  </ul>
</div>
`;
